feat: [SEISMO- 102, SEISMO-103, SEISMO-104] New Frontend Features SEISMO 5.0

L8zn opened the pull request:
<!--- Add title with the following format,

type: [SEISMO-XXXX] SHORTENED TASK TITLE

where type is:
1. draft - to be completed PR
2. feat - new feature
3. fix - bug fixes
4. test - unit tests
5. chore - (aka housekeeping) cleaning/styling/refactor code, documentations, adding comments
--> 

# Detailed Description

### Summary
<!--- Please link the related issue/task --> 
> This PR includes: [SEISMO 102 - Map Layers Button (Base + Overlays)](https://www.notion.so/jccabang/Seismo-102-25e494af83a080b18be4db89705a6d6a?source=copy_link), [SEISMO 103 - Legend Button (context-aware)](https://www.notion.so/jccabang/Seismo-103-25e494af83a080e9bf77fb94ee3afc38?source=copy_link), [SEISMO 104 - Advanced Search (left panel) + Presets](https://www.notion.so/jccabang/Seismo-104-25e494af83a080098f59ff1e37851820?source=copy_link)

<!--- Please include a detailed summary of the changes --> 
Added a Leaflet LayersControl to the main map with:
- Basemaps: OSM Standard, CartoDB Positron, CartoDB DarkMatter, Esri World Imagery (via leaflet-providers)
    <img width="387" height="618" alt="image" src="https://github.com/user-attachments/assets/3c494038-7a99-46b4-abd3-3bc25badd787" />
    
- Overlays (toggleable):
- Fault Lines (GEM Global Active Faults via CDN; filtered to Philippines bbox)
        
<img width="1209" height="1402" alt="image" src="https://github.com/user-attachments/assets/be00e866-90ae-4f28-b758-81473bf9fa17" />
    
- Plate Boundaries (PB2002 Bird 2003 via CDN)
     
<img width="1188" height="1392" alt="image" src="https://github.com/user-attachments/assets/788d05cb-4bb7-4c90-9b0c-07fa3df43828" />

 - Added a info tooltip popup for selected Fault Line and Plate Boundary overlays
    
<img width="1627" height="1395" alt="image" src="https://github.com/user-attachments/assets/8bd13dd7-581b-44f8-9120-cf5bf1a24e7a" />

- Attribution control shows provider attributions and fixed the desktop view attribution control position to be viewable.

<img width="663" height="208" alt="image" src="https://github.com/user-attachments/assets/48288799-9969-4780-a1bc-f8abbdeb2057" />

- Overlay performance: Remote GeoJSON is cached in-memory after first load; toggling doesn’t re-fetch or re-parse large files.
- All configuration is centralized and minimal. No local vector tile server is required for development.
- Built a custom L.Control legend that reads which overlay groups are ON and renders corresponding symbology (line color/width for faults/boundaries)
    - included data source/last update for Remote GeoJSON
   
<img width="537" height="478" alt="image" src="https://github.com/user-attachments/assets/86cb0489-bfb8-4a50-9979-e2bc7119dce3" />
    
- Restricted vertical bounds on leaflet map while horizontal panning is not blocked|

- Added a theme based styling based on the Basemap selected.

<img width="2559" height="1400" alt="image" src="https://github.com/user-attachments/assets/b05d27cb-80a5-4a0e-8c04-1a2945eae555" />

- Added a depth ramp for Earthquake Markers overlay

<img width="1902" height="1324" alt="image" src="https://github.com/user-attachments/assets/aac8a851-23fc-4908-b86c-f6f23413dd63" />

- Improved sidebar with added features
    - Built filter UI: time window, magnitude (min/max)
    - Added search filter UI
    - Added Presets (Latest EQs, 2025 EQs, 2024 EQs, 2023 EQs) that changes the shown EQ markers in the map based on selected preset
 
<img width="473" height="1397" alt="image" src="https://github.com/user-attachments/assets/e7b90005-4fe1-485b-b936-f7aed26f524a" />

- Made the changes responsive in mobile screens

<img width="596" height="1268" alt="image" src="https://github.com/user-attachments/assets/afcc88b1-c80d-49c6-9f8a-4a75de68f0ce" />


### Motivation & Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
The motivation of these changes is to improve the utilities of the current Web App for its users and refine the UI/UX design. 
### Dependencies
<!--- List any dependencies that are required for this change. -->
None 
### Type of change
<!--- Please delete options that are not relevant. -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

<!--- If breaking change, detail which existing functionality/s should or is expected to change -->

# Checklist:
<!--- Double check the following and leave uncheck those that you haven't done or are not applicable. -->

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

chrscbs committed:
feat: add react-router-dom module

chrscbs committed:
chore: utilize Router in App.js, move the map view content to /src/pages/HomePage.js)

chrscbs committed:
feat: add new endpoints in frontend; this pages will be utilized by the interns

uNique-02 committed:
[FEAT] Earthquake Info Page

uNique-02 committed:
Update Articles.js

L8zn committed:
Added frontend of significant-eq-page

chrscbs committed:
chore: fix import directories

chrscbs committed:
chore: rename significant eqs page css

chrscbs committed:
chore: rename css file

chrscbs committed:
feat: add axios call to backend to fetch all significant eqs

chrscbs committed:
feat: transition to earthquake info page on click of a specific card

chrscbs committed:
feat: add onClick function to the Card component

chrscbs committed:
Merge remote-tracking branch 'origin/significant-eq-page' into eq-info-page

chrscbs committed:
chore: clean EQ Info page files

chrscbs committed:
fix: refactor eq info page code for to properly display data from backend

chrscbs committed:
feat: add param to open in new tab each clickable link

chrscbs committed:
chore: refactor Articles component structure

chrscbs committed:
fix: redirect to UPRI seismic network FDSNWS each download link

chrscbs committed:
fix: add rel to href target _blank to remove the warning

chrscbs committed:
chore: add margin on big screens to EQ info page

chrscbs committed:
chore: make card title smaller

chrscbs committed:
chore: delete unused components

chrscbs committed:
fix: add -8 to the time of mseed files to get

chrscbs committed:
chore: change date format displayed

chrscbs committed:
feat: add floating action button component

chrscbs committed:
feat: display significant eqs floating action button on home page

chrscbs committed:
chore: add description to time

chrscbs committed:
chore: hardcode article source

chrscbs committed:
chore: adjustsection margin

chrscbs committed:
feat: adjust article card design to match with the card initial card component

chrscbs committed:
feat: adjust article card design to match with the card initial card component

chrscbs committed:
chore: change webpage description

chrscbs committed:
chore: remove get request to url, set the article info to default

chrscbs committed:
chore: parse event summary saved in db with html elements

chrscbs committed:
fix: properly display html elements from db

chrscbs committed:
feat: pass initial station list to display count of currently streaming devices

chrscbs committed:
Merge remote-tracking branch 'origin/eq-info-page' into online-devices-indicator

chrscbs committed:
chore: hide online stations count in header when not in home page

L8zn committed:
chore: freeze baseline prior to map UI features

L8zn committed:
chore: update .gitignore to exclude public/config.js

L8zn committed:
docs: map features tech design (frozen stack)

L8zn committed:
docs: add API, component, event flow, and state flow maps

L8zn committed:
docs: update MAP_FEATURES_DESIGN

L8zn committed:
feat(map): add LayersControl with OSM/Carto/Esri basemaps; CDN fault and plate overlays; styled headings; overlay caching

L8zn committed:
Fix: map hard‑stop vertical bounds, infinite horizontal pan

L8zn committed:
feat(map): context-aware legend panel; minor leaflet-control-attribution adjustment

L8zn committed:
chore: Enhance Map Layers Control and Overlay Functionality

- Improved accessibility and usability of the Map Layers Control by implementing click-based toggling, ARIA attributes, and keyboard shortcuts for layer management.
- Added focus trapping and improved focus management for the layers panel.
- Introduced a new toggleOverlay function in OverlayStateContext to manage layer visibility more effectively.
- Enhanced RemoteGeoJSONOverlay to support world copies of geometries by shifting longitudes.
- Updated StationMarker styles for better visibility and theme awareness.
- Revamped legend styles for improved aesthetics and responsiveness, including theme-aware adjustments.
- Refined map layers CSS for better layout and interaction, including hover effects and responsive design.
- Adjusted map layer styles for faults and plates to improve visual clarity.
- Established a consistent font and color scheme across the application for better readability and user experience.

L8zn committed:
chore(map): update map layers control titles and improve zoom button styling

L8zn committed:
fix: enhance theme detection for basemap layers using URL and attribution heuristics

L8zn committed:
feat(map): add interactive tooltips for faults and plates; enhance styling for feature hover

L8zn committed:
feat: implement centralized dataset management and last updated metadata for overlays

L8zn committed:
chore: Refactor map styling and overlays for improved theme support

- Introduced centralized map styling in `mapStyles.js` for earthquakes, faults, plates, and stations.
- Updated `EventMarkers.js` to filter events based on zoom level and added depth handling.
- Enhanced `LegendControl.js` to include depth ramp toggle and improved styling based on theme.
- Modified `MapLayersControl.js` to reflect active overlays and apply theme-aware styles.
- Updated `RemoteGeoJSONOverlay.js` to support dynamic styles and interaction.
- Refined CSS for station markers and legend components to utilize CSS variables for theme consistency.
- Removed overlay swatches from the Layers panel for a cleaner UI, relying on the legend for symbology.

L8zn committed:
feat: add search presets UI with sidebar updates

L8zn committed:
feat: enhance sidebar and event markers with filter support and SSE toggle

L8zn committed:
feat: update sidebar options to include 2023 earthquakes and temporarily remove major earthquakes filter

chrscbs commented:
@L8zn, painclude ng mga screenshots ng updates dito sa PR, for reference. Thanks!

chrscbs commented:
Test Done:
1. Imported all recorded events (650+ entries) from server to my local machine, loading takes  
  - <img width="1365" height="659" alt="image" src="https://github.com/user-attachments/assets/7488b894-73d0-43ee-91aa-95c942e0453f" /> _no problem with loading the frontend if no events entry is found in database_
  - <img width="1365" height="659" alt="image" src="https://github.com/user-attachments/assets/4412472e-3213-4f75-b455-1b4539698bf2" /> _upload successful of events entries on local mongodb_
  - <img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/a1689a3c-664d-43ab-897e-bff24f2b7c6d" /> _with 650+ entries in events collection, the frontend is stuck on loading screen_
  - I also tried inserting only 1 entry in events, but the loading screen is still stuck.

L8zn committed:
feat: enhance event fetching and state management in HomePage component

L8zn committed:
feat: enhance event handling in EventMarkers to update existing events

L8zn committed:
feat: improve popup handling in EventMarker for better map interaction

L8zn committed:
fix: enhance EventMarker to validate coordinates and improve popup handling
- solves bug caused by "Cannot read properties of undefined (reading 'lat')"

L8zn committed:
feat: optimize EventMarker icon rendering with memoization and caching

L8zn committed:
feat(ui): improve map and sidebar interactions

- made fault line strokes thinner, with hover scale-up for clarity
- fixed bug where toggling earthquake overlay showed all preset markers
- added empty-state indicator in SidebarItems for better feedback
- enhanced EventMarker fade-in when switching presets/overlays

L8zn committed:
feat: enhance EventMarker re-render logic and improve EventMarkers and HomePage for better event handling and state management
- fixed issue with showing all EQ markers when toggling off then on the EQ overlay

chrscbs commented:
Current updates now properly works on my end. Good to merge 🎉
